pydantic/json_schema.py

```diff
@@ -1201,9 +1200,7 @@ def _extract_discriminator(
                     continue  # this means that the "alias" does not represent a field
                 alias_is_present_on_all_choices = True
                 for choice in one_of_choices:
-                    while '$ref' in choice:
-                        assert isinstance(choice['$ref'], str)
-                        choice = self.get_schema_from_definitions(JsonRef(choice['$ref'])) or {}
+                    choice = self.resolve_schema_to_update(choice)
```

Comment:
This function handles the above removed logic more gracefully, with an intuitive error.